uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/ILogger.java

```diff
@@ -0,0 +1,122 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.logging;
+
+import androidx.annotation.NonNull;
+
+/**
+ * An interface for sending log messages.
+ */
+public interface ILogger {
```

Comment:
A bug question: Do we need another interface here with our own logging system?  I was kind of hoping that we could shift to logback or slf4j so that we could have better control over the emitted messages, and so that we could get logging out of our dependencies that already use java.util.logging or some other mechanism.

So if we're going down this path, for the most part, logging subsystems at this point tend to split their functionality into two different pieces.  You have a Logger, which is the interface used by the code and knows nothing about how the data is consumed.  This typically translates whatever various inputs the user gives to it into a message class, potentially filtered by a configuration, and then forwards that message to a series of appenders.  The appenders are what takes care of writing things to files (in this case, though it could be sending things on a socket), and they can be configured and controlled on a finer granularity than what we have here.

I'm just seeing us writing a lot of our own loggers (there's one in OneAuth, too).

Reply:
Added an interface for appenders.....currently all logs are routed to all (registered) appenders...we could refactor in the future to route specific logs to specific appenders based on tags or something else.....will keep that out of scope of this PR